### PR TITLE
feat(flags): expose subscriptions api behind feature flag for test harness

### LINKED ISSUE #109: add feature flagging
## Describe the problem you are facing 
Our primary api, the promise api, should be the only one exposed at this time imo. All other apis such as subscriptions should be hidden until they are ready for public consumption. 

## Describe the solution you'd like
I propose that we hide the experimental features behind feature flags. Users can enable the experimental features if they want, but they explicitly know not to expect a mature feature.

Looked at: 
- https://prometheus.io/docs/prometheus/latest/feature_flags/
- https://gofeatureflag.org/docs 
- https://github.com/open-feature

## Alternatives you've considered
Another alternative would be to just document them as experimental/unfinished features. Although, we should prob do both. 

## Additional context
n/a

### PR DESCRIPTION
**Pull request by guergabo**
Fixes https://github.com/resonatehq/resonate/issues/109

### CODE REVIEW

**Comment by guergabo** [comment]
remove api

File: cmd/serve.go [hunk 1]
```diff
@@ -162,6 +163,7 @@ var serveCmd = &cobra.Command{
 
 func init() {
 	// api
+	serveCmd.Flags().Var(&ExperimentalFeatures{}, "enable-feature", "enable features that are disabled by default since they are considered experimental")
```
Comment (dfarr) [inline cmd/serve.go:166]:
```suggestion
	serveCmd.Flags().Var(&ExperimentalFeatures{}, "enable-feature", "enable experimental features that are disabled by default")
```

File: cmd/serve.go [hunk 2]
```diff
@@ -226,3 +229,28 @@ func init() {
 	serveCmd.Flags().SortFlags = false
 	rootCmd.AddCommand(serveCmd)
 }
+
+type ExperimentalFeatures []string
```
Comment (dfarr) [inline cmd/serve.go:233]: could this move to the `flags.go` file?

File: internal/app/subsystems/api/http/http.go [hunk 3]
```diff
@@ -39,6 +42,13 @@ func New(api api.API, config *Config) api.Subsystem {
 	r.POST("/promises/:id/resolve", s.resolvePromise)
 	r.POST("/promises/:id/reject", s.rejectPromise)
 
+	// Subscription API (experimental)
+	if slices.Contains(config.Enable, "subscription") {
```
Comment (dfarr) [inline internal/app/subsystems/api/http/http.go:46]:
```suggestion
	if slices.Contains(config.Enable, "subscriptions") {
```

File: internal/app/subsystems/api/service/request.go [hunk 4]
```diff
@@ -58,3 +63,34 @@ type RejectPromiseHeader struct {
 type RejectPromiseBody struct {
 	Value promise.Value `json:"value"`
 }
+
+// Subscription
+
+type SearchSubscriptionsParams struct {
```
Comment (dfarr) [inline internal/app/subsystems/api/service/request.go:69]: We probably won't have a search subscriptions API, just a list all subscriptions on a promise (which is also paginated),

File: internal/app/subsystems/api/service/request.go [hunk 5]
```diff
@@ -58,3 +63,34 @@ type RejectPromiseHeader struct {
...
+	// Q      string `form:"q" json:"q"`
+	PromiseId string `json:"promiseId"`
+	Limit     int    `form:"limit" json:"limit"`
+	// Cursor    string `form:"cursor" json:"cursor"`
+}
+
+type CreateSubscriptionHeader struct {
+	RequestId      string                  `header:"request-id"` // qq: just for grpc?
```
Comment (dfarr) [inline internal/app/subsystems/api/service/request.go:77]: This is an optional parameter for tracking requests cross-systems. If provided this field is used in corresponding logs, if not provided I believe we generate a uuid for each request.

Comment (dfarr) [reply internal/app/subsystems/api/service/request.go:77]: you can set in both http/grpc

**Pull request closed without merging**